test/test_jsx.re

```diff
@@ -0,0 +1,567 @@
+      This forces the use of the wrapping functions provided in Xml.W.
+   */
+module HtmlWrapped: {
+  include
+    Html_sigs.T with
+      type Xml.elt = Tyxml.Xml.elt and type elt('a) = Html.elt('a);
+  include
+    LANGUAGE with
+      type elt('a) := elt('a) and
+      type wrap('a) := wrap('a) and
+      type list_wrap('a) := list_wrap('a) and
+      type doc := doc;
+} = {
+  include Html;
+  module Svg = Svg;
+};
+module HtmlWrappedTests = TyTests(HtmlWrapped);
+
+let (@:) = (h, t) => HtmlWrapped.Xml.W.(cons(return(h), t));
+let (@-) = HtmlWrapped.Xml.W.append;
+let nil = HtmlWrapped.Xml.W.nil;
+let (^) = HtmlWrapped.Xml.W.return;
+let (!:) = x => x @: nil();
+
+let wrapping = {
+  module Html = HtmlWrapped;
+  (
+    "wrapping",
+    HtmlWrappedTests.make(
+      Html.[
+        ("elem", !:[<p />], !:p(nil())),
+        ("child", !:[<p> <span /> </p>], !:p(span(nil()) @: nil())),
+        (
+          "list",
+          [<> <p /> <span> foo </span> </>],
+          p(nil()) @: span(txt("foo"^) @: nil()) @: nil(),
+        ),
+        ("attrib", !:[<p id=foo />], !:p(~a=[a_id("foo"^)], nil())),
+        (
+          "attribs",
+          !:[<p id=foo className=bar />],
+          !:p(~a=[a_id("foo"^), a_class(["bar"]^)], nil()),
+        ),
+        /* Can this be done with jsx?
+           ("comment", !:[%html "<!--foo-->"], !:(tot @@ Xml.comment("foo"))),
+           */
+        ("txt", !:[<p> foo </p>], !:p(txt("foo"^) @: nil())),
+        (
+          "wrapped functions",
+          !:[<input method=get />],
+          !:input(~a=[a_method(`Get^)], ()),
+        ),
+      ],
+    ),
+  );
+};
+
+let elt1 = () => !:HtmlWrapped.(span(!:txt("one"^)));
+let elt2 = () => !:HtmlWrapped.(b(!:txt("two"^)));
+let id = "pata"^;
+
+let antiquot = {
+  module Html = HtmlWrapped;
+  (
+    "ppx antiquot",
+    HtmlWrappedTests.make(
+      Html.
+        [
+          ("child", !:[<p> elt1 () </p>], !:p(elt1())),
+          ("list child", !:[<p> elt2 () </p>], !:p(elt2())),
+          (
+            "children",
+            !:[<p> bar elt1 () "foo" elt2 () baz </p>],
+            !:
+              p(
+                txt("bar"^)
+                @: elt1()
+                @- txt("foo"^)
+                @: elt2()
+                @- txt("baz"^)
+                @: nil(),
+              ),
+          ),
+          (
+            "insertion",
+            !:[<p> <em> elt1 () </em> </p>],
+            !:p(!:em(elt1())),
+          ),
+          (
+            "attrib",
+            !:[<p id> bla </p>],
+            !:p(~a=[a_id(id)], !:txt("bla"^)),
+          ),
+          ("first child", [(elt1())(<p />)], elt1() @- p(nil()) @: nil()),
+          ("last child", [<> <p /> {elt1()} </>], p(nil()) @: elt1()),
+          (
+            "wrapped functions",
+            !:[<input method=`Get />],
+            !:input(~a=[a_method(`Get^)], ()),
+          ),
+        ],
+        /* should succeed */
```

Comment:
what's wrong with this one?

Reply:
It was commented out preciously, I can just add it in I guess